logicrunner/logicrunner.go

```diff
@@ -232,127 +207,154 @@ func (lr *LogicRunner) Execute(ctx context.Context, inmsg core.SignedMessage) (c
 	if err != nil {
 		return nil, errors.Wrap(err, "Can't create request")
 	}
-	logicContext := core.LogicCallContext{
+
+	es.callContext = &core.LogicCallContext{
 		Caller:  msg.GetCaller(),
+		Callee:  &ref,
 		Request: reqref,
 		Time:    time.Now(), // TODO: probably we should take it from e
 		Pulse:   *lr.pulse(),
 	}
 
 	switch m := msg.(type) {
 	case *message.CallMethod:
-		re, err := lr.executeMethodCall(logicContext, m, vb)
+		re, err := lr.executeMethodCall(es, m, vb)
 		return re, err
 
 	case *message.CallConstructor:
-		re, err := lr.executeConstructorCall(logicContext, m, vb)
+		re, err := lr.executeConstructorCall(es, m, vb)
 		return re, err
 
 	default:
 		panic("Unknown e type")
 	}
 }
 
-func (lr *LogicRunner) pulse() *core.Pulse {
-	pulse, err := lr.Ledger.GetPulseManager().Current()
-	if err != nil {
-		panic(err)
-	}
-	return pulse
-}
-
+// ObjectBody is an inner representation of object and all it accessory
+// make it private again when we start it serialize before sending
 type ObjectBody struct {
-	Object core.ObjectDescriptor
-	Class  core.ClassDescriptor
-	Code   core.CodeDescriptor
+	objDescriptor   core.ObjectDescriptor
+	Object          []byte
+	ClassHeadRef    *Ref
+	CodeMachineType core.MachineType
+	CodeRef         *Ref
```

Comment:
I don't say this should be reverted, but we should aim different direction. I believe we should use more of Descriptors objects in our code and make sure they are passive carriers of data with declared interfaces. As we move forward towards new features we need more and more fields from each structure. My opinion is quite strong that ObjectBody should go away as a struct and instead we should keep separate fields for object, class and code. The last two can be pointers as we can be executing multiple objects with one class and thus the code, so we can share descriptors and minimize net travels for ledger accesses.

Do you agree?

Reply:
Main problem is we can't transfer it to another machine and reconstruct it there, so we need something transferable.